src/Sniffer.php

```diff
@@ -0,0 +1,278 @@
+        $sniffer->config = $this->prepareConfig($usingExplicitStandard, $lintLanguage, $fixing);
+        $sniffer->init();
+        $sniffer->reporter = new Reporter($sniffer->config);
+
+        // Find all the relevant code blocks for linting
+        // $files = $this->findMarkdownFiles($sniffer->config, $sniffer->ruleset);
+        if (PHP_CODESNIFFER_VERBOSITY > 0) {
+            echo 'Finding markdown files... ';
+        }
+
+        $files = new FileList($sniffer->config, $sniffer->ruleset);
+
+        if (PHP_CODESNIFFER_VERBOSITY > 0) {
+            $numFiles = count($files);
+            echo "DONE ($numFiles files in queue)" . PHP_EOL;
+        }
+
+        $codeBlocks = $this->findFencedCodeblocks($files, $lintLanguage);
+
+        // Add code blocks to the file list for linting
+        $todo = [];
+        foreach ($codeBlocks as $block) {
+            $dummy = new CodeBlock($block['content'], $sniffer->ruleset, $sniffer->config);
+            $dummy->num = $block['num'];
+            $dummy->path = $block['path'];
+            $dummy->realPath = $block['realpath'];
+            $todo[] = $dummy;
+        }
+
+        // Do the actual linting
+        $numErrors = $this->sniff($sniffer, $todo);
+
+        // The dummy files have the fixed content stored - but we still need to write that to the original files.
+        // There's no good AST to markdown renderer for league/commonmark so we're just doing a bit of an ugly
+        // search and replace.
+        if ($fixing) {
+            /** @var CodeBlock $dummy */
+            foreach ($todo as $dummy) {
+                if ($dummy->getFixedCount() < 1) {
+                    continue;
+                }
+
+                if (!is_file($dummy->realPath)) {
+                    // 3 is the exit code phpcs uses for errors like this
+                    throw new DeepExitException("Can't find file {$dummy->realPath} to set new content", 3);
+                }
+
+                /** @var FencedCode $mdBlock */
+                $mdBlock = $codeBlocks[$dummy->path]['md'];
+                $indent = str_repeat(' ', $mdBlock->getOffset());
+                $origBlockContent = preg_replace('/^/m', $indent, $mdBlock->getLiteral());
+                $newBlockContent = preg_replace('/^/m', $indent, preg_replace('/\s*<\?php\n?/', '', $dummy->getContent()));
+                $newFileContent = str_replace($origBlockContent, $newBlockContent, file_get_contents($dummy->realPath));
+
+                file_put_contents($dummy->realPath, $newFileContent);
+            }
+        }
+
+        $sniffer->reporter->printReports();
+
+        if ($numErrors === 0) {
+            // No errors found.
+            return 0;
+        } else if ($sniffer->reporter->totalFixable === 0) {
+            // Errors found, but none of them can be fixed by PHPCBF.
+            return 1;
+        } else {
+            // Errors found, and some can be fixed by PHPCBF.
+            return 2;
+        }
```

Comment:
See [`Runner::runPHPCS()`](https://github.com/squizlabs/PHP_CodeSniffer/blob/321d252dc9c180a085ecb6acc8a861aba8c4da2d/src/Runner.php#L137-L146)

Reply:
Should make this an inline comment about this block

Reply:
Done